Feature/dcg 142 sytemuser systemrole

Cesardf1 opened the pull request.

Cesardf1 committed:
Criação do fluxo Role
Conexão com o Backend funcional

Cesardf1 committed:
Tirada uma função inutil
A pagina agora vai atualizar as informações assim que entra nela

Cesardf1 committed:
Criado fluxo System_Permission

Cesardf1 committed:
Finalizando detalhes finais do CRUD do SystemUser antes de fazer o filtro

Cesardf1 committed:
Filtro funcionando.
Fluxo SystemUserRole criado e funcional.
Iniciado o fluxo SystemUserPermission.
Modal adicionado a tabela de relacionamento SystemUserRole.
Adicionado Common Module no projeto.
Modal mostrando os Roles existentes.

Cesardf1 committed:
Seleção de Roles por Checkbox criada.

Cesardf1 committed:
Alterada a entidade SystemUser para conter um array de SystemRoles.
Iniciado a criação das funções utilizadas pelo popup

Cesardf1 committed:
Alterada a entidade SystemUser para conter um array de SystemRoles.
Inicio da criação das funções necessárias no popup

pedroLFaria commented:
Ignorar todos os arquivos em .vs e deletar eles da branch.

pedroLFaria commented on src/app/core/usecases/system-permission/post-bank.usecase.ts:
```diff
@@ -0,0 +1,19 @@
+@Injectable({
+  providedIn: 'root',
+})
+export class PostSystemPermissionUsecase
```
Nome do arquivo está post-bank, diferente da classe

pedroLFaria replied:
Nome da classe diferente do nome do arquivo

pedroLFaria commented on src/app/core/usecases/system-role/post-bank.usecase.ts:
```diff
@@ -0,0 +1,17 @@
+@Injectable({
+  providedIn: 'root',
+})
+export class PostSystemRoleUsecase implements UseCase<SystemRoleModel, SystemRoleModel> {
```
Outro arquivo com nome post-bank, isso está certo?

pedroLFaria commented on src/app/data/lookup-repository/cargo-sistema-lookup-web.repository.ts:
```diff
@@ -0,0 +1,52 @@
+@Injectable({
+  providedIn: 'root',
+})
+export class SystemRoleLookupWebRepository extends SystemRoleLookupRepository {
```
Nome do arquivo e nome da classe estão bem diferentes e idiomas diferentes, está certo isso?

pedroLFaria commented on src/app/data/lookup-repository/permissao-sistema-lookup-web.repository.ts:
```diff
@@ -0,0 +1,52 @@
+@Injectable({
+  providedIn: 'root',
+})
+export class SystemPermissionLookupWebRepository extends SystemPermissionLookupRepository {
```
Nome da classe e arquivo estão bem diferentes, pode checar isso?

Cesardf1 committed:
Alteração de nome de variável para ficar mais claro o que é.
Alteração no mock de dados para permitir testes melhores.
Alteração na função "handleValueChanged" que agora obtém os resultados esperados.

pedroLFaria commented on src/app/view/pages/system-user-role/system-user-role.component.ts:
```diff
@@ -71,30 +70,42 @@ export class SystemUserRoleComponent implements OnInit {
-          this.teste = true;
-        }
+
+    for (j = 0; j < this.selectedSystemUser.systemRoles.length; j++) {
```
Verificar outros tipos de for que temos no JS, como for in e for on

Cesardf1 committed:
.vs adicionado ao git ignore.
conteudo do .vs deletado.

Cesardf1 commented:
@pedroLFaria @Cesardf1
.vs adicionado ao git ignore e
 conteudo do .vs deletado.

Cesardf1 committed:
Value no html recebe o resultado da nova função setValue
para definir se o checkbox deve ou não estar marcado
quando o modal é aberto.
Função handleValueChanged alterada para adicionar as roles
marcadas no checkbox no array de roles do usuário.

pedroLFaria commented on src/app/view/pages/system-user-role/system-user-role.component.ts:
```diff
@@ -42,6 +41,14 @@ export class SystemUserRoleComponent implements OnInit {
 
+  setValue(role: SystemRoleModel) {
+    var i = 0;
+    for (i = 0; i < this.selectedSystemUser.systemRoles.length; i++) {
```
Ver os outros tipos de for no JS ou as funções que iteram tbm.

pedroLFaria commented on src/app/view/pages/system-user-role/system-user-role.component.ts:
```diff
@@ -94,24 +101,22 @@ export class SystemUserRoleComponent implements OnInit {
-      }
+    if (e.value == true) {
+      this.selectedSystemUser.systemRoles.push(role);
+    } else {
```
Isso é um else vazio?

Cesardf1 committed:
Resolvendo os comentários do pull request

Cesardf1 committed:
Resolvendo os comentarios do pull request

Cesardf1 committed:
Alterados o model,web-entity e repositorio de SystemUser para ficarem de acordo com os padrões esperados.
Condição adicionada ao html antes de mostrar a checkbox.
SelectedSystemUser não mais alterando diretamente o datasource.
Função removeRoleById criada.
Função addInDataSource adicionando as alterações feitas ao DataSource.
Função applyChanges chama a função addInDataSource e fecha o modal.

Cesardf1 committed:
Criado fluxo de envio para o backend do SystemUserSystemRole

Cesardf1 committed:
Criado o fluxo para o backend do SystemUserSystemRole

Cesardf1 committed:
corrigido os environments utilizados no SystemUserSystemRoleManageWebRepository
